### ISSUE TITLE
Any plans to include `sel` in stackage?

### ISSUE THREAD

**Issue by Daniel-Diaz** [haskell-cryptography/libsodium-bindings#160]
Thank you for these great bindings!

Is there any chance they will be included in future stackage snapshots?

**Comment by Kleidukos** [comment]
@Daniel-Diaz Hi! I have no idea how these things work, unfortunately.

**Comment by Daniel-Diaz** [comment]
All that's needed is for the package to be included here: https://github.com/commercialhaskell/stackage/blob/master/build-constraints.yaml

**Comment by Daniel-Diaz** [comment]
There's more info in the readme: https://github.com/commercialhaskell/stackage/tree/master?tab=readme-ov-file#add-your-package

**Comment by Kleidukos** [comment]
Hmpm, https://docs.haskellstack.org/en/stable/travis_ci/ and https://packdeps.haskellers.com/ are unusable. 

I'm not against the packages being present on Stackage, nor against accommodating Stackage by using good dependency bounds, but I am already spread so thin that this is not something I can do myself.

If you have a real need for `sel` and `libsodium-bindings` to be present on Stackage, you have my blessing to add them.

**Comment by Daniel-Diaz** [comment]
> Hmpm, https://docs.haskellstack.org/en/stable/travis_ci/ and https://packdeps.haskellers.com/ are unusable.
> 
> I'm not against the packages being present on Stackage, nor against accommodating Stackage by using good dependency bounds, but I am already spread so thin that this is not something I can do myself.
> 
> If you have a real need for `sel` and `libsodium-bindings` to be present on Stackage, you have my blessing to add them.

Yeah, it seems their readme might be out-of-date. I'll take a look.

**Comment by Daniel-Diaz** [comment]
I needed this again, so I decided to take action this time. I would like to have the following packages added to stackage:

- sel
- libsodium-bindings
- text-display

Do I still have your blessing to do this? If so, I commit to:

- Adding the above packages under my name to stackage so that I get any pertinent notifications.
- Upon receiving any notification, opening an issue on the original package repository.
- If time permits, submitting a PR fixing the issue.

Let me know whether I have green light.

Thank you.

**Comment by Kleidukos** [comment]
Absolutely, you have my green light!

**Comment by Daniel-Diaz** [comment]
For reference: https://github.com/commercialhaskell/stackage/pull/7706

**Comment by Daniel-Diaz** [comment]
@Kleidukos 

The following dependency bounds (in Hackage) are too restrictive:

- `text-display ^>=0.0`
- `hedgehog ^>=1.4`

This prevents the package from building with the current versions of these packages, which blocks it from being added to the next stackage snapshot.

The bounds have already been updated in the `main` branch. Simply releasing that or submitting a revision would fix this. A new release would allow `sel` to still build against older versions of text-display and hedgehog. A revision with a wider version tolerance could also achieve this.

These actions are beyond my control, so I let you decide.

Thanks.

**Comment by Kleidukos** [comment]
Excellent I'll do this today

**Comment by Kleidukos** [comment]
@Daniel-Diaz Sorry for the time it took. Is it better now?

**Comment by Daniel-Diaz** [comment]
> [@Daniel-Diaz](https://github.com/Daniel-Diaz) Sorry for the time it took. Is it better now?

Thanks! I think that will work.

**Comment by Daniel-Diaz** [comment]
Sorry, I did find an issue.

The dependency on `text-display` was upgraded to `1.*`, but the version of the code on Hackage (`0.0.2.0`) doesn't actually build against this range, due to the change on the output type of `displayBuilder`. Unfortunately, this renders the package unbuildable.

**Comment by Kleidukos** [comment]
@Daniel-Diaz Alright, I'll revert the metadata changes on Hackage. Maybe Stackage should target the next major version?

**Comment by Daniel-Diaz** [comment]
> [@Daniel-Diaz](https://github.com/Daniel-Diaz) Alright, I'll revert the metadata changes on Hackage. Maybe Stackage should target the next major version?

I think that's a good plan.

**Comment by Daniel-Diaz** [comment]
I think we can close this now, as `sel` is already in stackage. Thank you!
